extensions/integration-tests/src/schemaCompare.test.ts

```diff
@@ -173,6 +174,15 @@ export function assertSchemaCompareResult(schemaCompareResult: azdata.SchemaComp
 }
 
 export async function assertScriptGenerationResult(resultstatus: azdata.ResultStatus, filepath: string): Promise<void> {
+	// TODO add more validation
 	assert(resultstatus.success === true, `Expected: success true Actual: "${resultstatus.success}" Error Message: "${resultstatus.errorMessage}`);
-	// TODO : validate script based on new return params
+	let retry = 10; // file takes quite long time to get created
+	let exists = false;
+	while (retry > 0 && !exists) {
+		exists = existsSync(filepath);
+		await utils.sleep(2000);
+		retry--;
+	}
+	assert(exists, `script file ${filepath} is expected to be present`);
```

Comment:
The process might still be writing to the file right? That might cause problems if you're deleting the file but it's still being written to. Is there a hook somewhere that you can have the test use for when schemaCompare is done generating its script?

Reply:
yeah wanted the same, but don't have. (in favor of completing this PR) I am leaning towards removing the script check for now since its cheeked from sqltools service so checking success is enough for now. will add deeper checks later

Reply:
yeah wanted the same but there isn't.

Reply:
also, there won't be a file to check for anymore after the change to open the generated script goes in.

Reply:
Ah more than happy to remove this (caused most issues) :). Will just validate status for now. Will add any more checks after your change.